recipes/cli11/all/conanfile.py

```diff
@@ -12,20 +13,40 @@ class CLI11Conan(ConanFile):
     license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/CLIUtils/CLI11"
-    topics = "cli-parser", "cpp11", "no-dependencies", "cli", "header-only"
+    topics = "cli-parser", "cpp11", "no-dependencies", "cli"
     package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
+    options = {
+        "header_only": [True, False]
+    }
+    default_options = {
+        "header_only": False
```

Comment:
That really sucks because header only builds are very undesirable. It ends up breaking unrelated code due to leaked implementation details like included system headers. That's what prompted me to create the issue and PR in the first place.

Reply:
The main issue here is that upstream also defaults to a single-header installation, and the compilation behaviour is opt-in, so with this we would be following the choices on upstream

Reply:
It's a very user hostile default though, only motivated by ease of distribution, which is hardly a concern if one is using Conan.

Reply:
@vasama You still can build using the option that you need in your configuration as well. In CCI we mostly follow upstream default behavior, as it will affect all users and will reflect the upstream build as well.

Reply:
I'm aware of that. I just want to make it clear that header only is a bad and pointless default.